flink-table-store-spark/src/main/java/org/apache/flink/table/store/spark/SparkCatalog.java

```diff
@@ -284,14 +284,19 @@ private UpdateSchema toUpdateSchema(
                                 }));
         Map<String, String> normalizedProperties = new HashMap<>(properties);
         normalizedProperties.remove(PRIMARY_KEY_IDENTIFIER);
+        String pkAsString = properties.get(PRIMARY_KEY_IDENTIFIER);
+        List<String> primaryKeys =
```

Comment:
We can add this check in `SchemaManager`.
- Primary / partition keys should in fields.
- Pk not null.